include/pros/ext_adi.h

```diff
@@ -0,0 +1,641 @@
+ * Gets the current gyro angle in tenths of a degree. Unless a multiplier is
+ * applied to the gyro, the return value will be a whole number representing
+ * the number of degrees of rotation times 10.
+ *
+ * There are 360 degrees in a circle, thus the gyro will return 3600 for one
+ * whole rotation.
+ *
+ * This function uses the following values of errno when an error state is
+ * reached:
+ * ENXIO - The given value is not within the range of ADI Ports
+ * EADDRINUSE - The port is not configured as a gyro
+ *
+ * \param gyro
+ *        The adi_gyro_t object for which the angle will be returned
+ *
+ * \return The gyro angle in degrees.
+ */
+double ext_adi_gyro_get(ext_adi_gyro_t gyro);
+
+/**
+ * Initializes a gyroscope on the given port. If the given port has not
+ * previously been configured as a gyro, then this function starts a 1300 ms
+ * calibration period.
+ *
+ * It is highly recommended that this function be called from initialize() when
+ * the robot is stationary to ensure proper calibration.
+ *
+ * This function uses the following values of errno when an error state is
+ * reached:
+ * ENXIO - The given value is not within the range of ADI Ports
+ * EADDRINUSE - The port is not configured as a gyro
+ *
+ * \param adi_port
+ *        The ADI port to initialize as a gyro (from 1-8, 'a'-'h', 'A'-'H')
+ * \param smart_port
+ *        The smart port number that the ADI Expander is in
+ * \param multiplier
+ *        A scalar value that will be multiplied by the gyro heading value
+ *        supplied by the ADI
+ *
+ * \return An adi_gyro_t object containing the given port, or PROS_ERR if the
+ * initialization failed.
+ */
+ext_adi_gyro_t ext_adi_gyro_init(uint8_t adi_port, uint8_t smart_port, double multiplier);
+
+/**
+ * Resets the gyroscope value to zero.
+ *
+ * This function uses the following values of errno when an error state is
+ * reached:
+ * ENXIO - The given value is not within the range of ADI Ports
+ * EADDRINUSE - The port is not configured as a gyro
+ *
+ * \param gyro
+ *        The adi_gyro_t object for which the angle will be returned
+ *
+ * \return 1 if the operation was successful or PROS_ERR if the operation
+ * failed, setting errno.
+ */
+int32_t ext_adi_gyro_reset(ext_adi_gyro_t gyro);
+
+/**
+ * Disables the gyro and voids the configuration on its port.
+ *
+ * This function uses the following values of errno when an error state is
+ * reached:
+ * ENXIO - The given value is not within the range of ADI Ports
+ * EADDRINUSE - The port is not configured as a gyro
+ *
+ * \param gyro
+ *        The adi_gyro_t object to be shut down
+ *
+ * \return 1 if the operation was successful or PROS_ERR if the operation
+ * failed, setting errno.
+ */
+int32_t ext_adi_gyro_shutdown(ext_adi_gyro_t gyro);
+
+
+#define SMART_PORT_BITS 16
+#define SMART_PORT_MASK ((1 << SMART_PORT_BITS) - 1)
+
+  /**
+   * Macro Description: Given a port mask, it sets the smart port and adi port to the values masked inside the int_32. 
```

Comment:
Minuscule nit: "macro description" is redundant and it's `int32_t` not `int_32`.